File: dotnet/src/Wata.Sdk/Acquiring/Models/RefundRequest.cs
namespace Wata.Sdk.Acquiring.Models;

/// <summary>
/// Запрос на возврат. Валюта берётся из исходной транзакции, поля "причина" не
/// существует — это осознанное ограничение API, а не упущение SDK.
/// </summary>
public sealed record RefundRequest(Guid OriginalTransactionId, decimal Amount);

File: java/src/main/java/pro/wata/sdk/model/RefundRequest.java
package pro.wata.sdk.model;

/**
 * Запрос на возврат ({@code POST /api/h2h/transactions/refunds}). Валюта
 * берётся из исходной транзакции — отдельного поля для неё нет, как и поля
 * «причина возврата».
 *
 * @param originalTransactionId идентификатор исходной транзакции (uuid), должна быть в статусе {@code Paid}
 * @param amount                сумма возврата: больше нуля, не больше доступного остатка, не более двух знаков после запятой
 */
public record RefundRequest(String originalTransactionId, double amount) {
}
